### ISSUE TITLE
[REVIEW]: txshift: Efficient estimation of the causal effects of stochastic interventions in R

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2447]
**Submitting author:** @nhejazi (<a href="http://orcid.org/0000-0002-7127-2789">Nima Hejazi</a>)
**Repository:** <a href="https://github.com/nhejazi/txshift" target ="_blank">https://github.com/nhejazi/txshift</a>
**Version:** v0.3.4-joss
**Editor:** @marcosvital
**Reviewer:** @klmedeiros, @joethorley
**Archive:** <a href="https://doi.org/10.5281/zenodo.4070043" target="_blank">10.5281/zenodo.4070043</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/e117eca30ec498ac5976dcd7636e642a/status.svg)](https://joss.theoj.org/papers/e117eca30ec498ac5976dcd7636e642a)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/e117eca30ec498ac5976dcd7636e642a"><img src="https://joss.theoj.org/papers/e117eca30ec498ac5976dcd7636e642a/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/e117eca30ec498ac5976dcd7636e642a/status.svg)](https://joss.theoj.org/papers/e117eca30ec498ac5976dcd7636e642a)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/nhejazi/txshift" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@klmedeiros & @joethorley, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @marcosvital know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/nhejazi/txshift">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nhejazi) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/nhejazi/txshift">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@nhejazi) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @klmedeiros, @joethorley it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.2202/1557-4679.1217 is OK
- 10.1111/j.1541-0420.2011.01685.x is OK
- 10.1002/sim.5907 is OK
- 10.1007/978-3-319-65304-4_14 is OK
- 10.1007/978-1-4419-9782-1 is OK
- 10.1007/978-3-319-65304-4 is OK
- 10.2202/1557-4679.1356 is OK
- 10.1111/rssb.12362 is OK
- 10.5281/zenodo.1342293 is OK
- 10.5281/zenodo.3558313 is OK
- 10.5281/zenodo.3698329 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02447/joss.02447/10.21105.joss.02447.pdf)

**Comment by marcosvital** [comment]
Dear @nhejazi and @benkeser: your manuscript will be reviewed in this issue, and you can reply any comments and suggestions that the reviewers might address right here.

**Comment by marcosvital** [comment]
@klmedeiros and @joethorley: thank you all for for accepting review this submission for JOSS.

Even if you are not starting the review right now, please accept the invite, as it has an expiration date (there is a link under Reviewer instructions & questions and you should also get an email notification). Furthermore, please check the instructions and checklists above, and let me know if you need any assistance.

You can also tag @nhejazi and @benkeser if you need to ask specific questions about the submission.

**Comment by joethorley** [comment]
@marcosvital can you clarify what JOSS means by "clear guidelines for third parties wishing to Seek support". 

See https://github.com/nhejazi/txshift/issues/57

I personally think the "Seek support" condition is covered by "Report issues or problems with the software" but please advise if this is not the case.

**Comment by marcosvital** [comment]
Hi, @joethorley. I agree: having a clear way to report issues or problems is enough.

But now that you mentioned it, I think that a small modification could be helpful: users should be pointed to CONTRIBUTING.md, since there are detailed instructions on how to file an issue there. Currently, if someone reads the issues statement alone, it's not possible to know that there are more instructions under contributing guidelines. @nhejazi, does that make sense?

**Comment by nhejazi** [comment]
Hi @marcosvital and @joethorley, thanks for the discussion on report issues. I've made the change to point users to further instructions in the `CONTRIBUTING.md` in the "Issues" section of `README.Rmd` (see https://github.com/nhejazi/txshift/pull/60). I'll be using that PR to keep track of all revisions and will merge it once we've resolved all comments/issues raised in this thread.

**Comment by joethorley** [comment]
@nhejazi - I just want to make sure you haven't forgotten about this issue https://github.com/nhejazi/txshift/issues/59

**Comment by joethorley** [comment]
@nhejazi and @marcosvital - I've completed my review and I am of the opinion that the authors have satisfied all the requirements for publication in JOSS.

**Comment by nhejazi** [comment]
Thanks very much for the detailed review @joethorley, it’s much appreciated.

@klmedeiros, just a friendly reminder about this review whenever you have some time open up.

**Comment by marcosvital** [comment]
Hi, @klmedeiros.
Can you give us an update about the review? Please let us know if you think you won't be able to review anymore, ok?

**Comment by klmedeiros** [comment]
@marcosvital my sincerest apologies, this slipped my mind. i'm going to work on it today with hopes to have it reviewed by tomorrow.

**Comment by marcosvital** [comment]
No problem, @klmedeiros!

**Comment by nhejazi** [comment]
@klmedeiros, sorry to bump this again. If there's anything I can do to help facilitate the review process, please don't hesitate to let me know.

**Comment by klmedeiros** [comment]
@nhejazi  My only reservations currently are that I could only get `txshift` to install by installing `haldensify ` from Github, not CRAN, which isn't documented anywhere. 

Otherwise, this is excellent and I recommend for publication in JOSS.

**Comment by nhejazi** [comment]
Thanks for the quick reply @klmedeiros. That's a great catch, I had missed that I had (at some point recently, I guess) bumped the version of the `haldensify` to 0.0.6 in `txshift` and forgotten to submit that minor release of `haldensify` to CRAN. I've just submitted that CRAN update and will ping here again once 0.0.6 of `haldensify` is on CRAN. This installation error should be taken care of then.

**Comment by nhejazi** [comment]
Quick update: `haldensify` v0.0.6 is now available on CRAN, so the installation issue should be resolved. I've also made a few changes for CRAN submission of the package, which is now in process. If the review is complete, I'll go ahead and merge https://github.com/nhejazi/txshift/pull/60. Are we all set @joethorley @klmedeiros @marcosvital?

**Comment by joethorley** [comment]
@nhejazi - I am

**Comment by klmedeiros** [comment]
@nhejazi In my opinion, yes! g2g

**Comment by nhejazi** [comment]
Excellent. Merged https://github.com/nhejazi/txshift/pull/60 to `master`. @whedon generate pdf

**Comment by nhejazi** [comment]
@whedon generate pdf

**Comment by nhejazi** [comment]
@marcosvital, whenever convenient, ready on my end to move on to final steps.

**Comment by marcosvital** [comment]
Hi, @nhejazi! Sorry about the delayed reply. Since all the reviewers are satisfied, we can carry on.

@klmedeiros and @joethorley, thank you very much for the time and effort put into this review!

**Comment by marcosvital** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2202/1557-4679.1217 is OK
- 10.1111/j.1541-0420.2011.01685.x is OK
- 10.1002/sim.5907 is OK
- 10.1007/978-3-319-65304-4_14 is OK
- 10.1007/978-1-4419-9782-1 is OK
- 10.1007/978-3-319-65304-4 is OK
- 10.2202/1557-4679.1356 is OK
- 10.1111/rssb.12362 is OK
- 10.1111/biom.13375 is OK
- 10.5281/zenodo.1342293 is OK
- 10.5281/zenodo.3558313 is OK
- 10.5281/zenodo.3698329 is OK

MISSING DOIs

- 10.21105/joss.02526 may be a valid DOI for title: hal9001: Scalable highly adaptive lasso regression in R

INVALID DOIs

- None
```

**Comment by marcosvital** [comment]
@nhejazi, can you check the missing DOI?

**Comment by nhejazi** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2202/1557-4679.1217 is OK
- 10.1111/j.1541-0420.2011.01685.x is OK
- 10.1002/sim.5907 is OK
- 10.1007/978-3-319-65304-4_14 is OK
- 10.1007/978-1-4419-9782-1 is OK
- 10.1007/978-3-319-65304-4 is OK
- 10.2202/1557-4679.1356 is OK
- 10.1111/rssb.12362 is OK
- 10.1111/biom.13375 is OK
- 10.21105/joss.02526 is OK
- 10.5281/zenodo.1342293 is OK
- 10.5281/zenodo.3558313 is OK
- 10.5281/zenodo.3698329 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by nhejazi** [comment]
@marcosvital, no worries about the delay and thanks for catching the DOI issue. It's now fixed. Would you mind updating the package version in the first comment on this thread from 0.3.3 to 0.3.4? (I can't seem to do that myself.)

**Comment by nhejazi** [comment]
@whedon build pdf

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by nhejazi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: :point_left:

**Comment by marcosvital** [comment]
@whedon set v0.3.4 as version

**Comment by whedon** [comment]
OK. v0.3.4 is the version.

**Comment by marcosvital** [comment]
@nhejazi, I'll read the last proof you generated, and will let you know if I find any issues. Please also take a careful (and probably final) look at it, ok? We are almost finished!

**Comment by nhejazi** [comment]
Thanks @marcosvital, I gave the paper another read through and made some minor corrections to improve its readability. I think I'm all set with it, but happy to correct any issues you might find. I'll rebuild the latest proof below.

**Comment by nhejazi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: :point_left:

**Comment by nhejazi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: :point_left:

**Comment by nhejazi** [comment]
Ok, current draft LGTM

**Comment by marcosvital** [comment]
Hi, @nhejazi! Almost there, but I would like to ask for two modifications:
 - please include the "statement of need" a separate paragraph, it should look [like this](https://github.com/openjournals/joss-papers/blob/joss.02584/joss.02584/10.21105.joss.02584.pdf) or [this](https://joss.theoj.org/papers/10.21105/joss.02342);
 - can you try start the summary with a more general, less technical explanation? JOSS guidelines for the summary asks for a "description of the high-level functionality and purpose of the software for a diverse, non-specialist audience". Usually, this can be easily achieved with one or two phrases (and I think that both papers linked above are good examples for that).

**Comment by nhejazi** [comment]
Thanks, @marcosvital! I think the requested modifications are good points, and I've added both a very short introductory statement and a statement of need similar to those appearing in the examples. Please note that this package provides access to very sophisticated statistical methodology, so I'm not sure that I can write a more general introduction than the one now available.

**Comment by nhejazi** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02447/joss.02447/10.21105.joss.02447.pdf) :page_facing_up: :point_left:

**Comment by nhejazi** [comment]
This version also LGTM. Please let me know of any further corrections/recommendations, @marcosvital.

**Comment by marcosvital** [comment]
Looks great to me, @nhejazi, so let's proceed. You will need to archive the last release of the package (on Zenodo, figshare, or other) if you already didn't do this. After that, just let me know the archive DOI. Once this is done, we'll be ready to publish.

**Comment by nhejazi** [comment]
Great! Thanks for the quick confirmation @marcosvital. I've created a new release at https://github.com/nhejazi/txshift/releases/tag/v0.3.4-joss and a Zenodo archive of that release at https://zenodo.org/record/4070043. The DOI for that release is 10.5281/zenodo.4070043.

**Comment by marcosvital** [comment]
@whedon set 10.5281/zenodo.4070043 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4070043" target="_blank">10.5281/zenodo.4070043</a> is the archive.

**Comment by marcosvital** [comment]
@whedon set v0.3.4-joss as version

**Comment by whedon** [comment]
OK. v0.3.4-joss is the version.

**Comment by marcosvital** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.2202/1557-4679.1217 is OK
- 10.1111/j.1541-0420.2011.01685.x is OK
- 10.1002/sim.5907 is OK
- 10.1007/978-3-319-65304-4_14 is OK
- 10.1007/978-1-4419-9782-1 is OK
- 10.1007/978-3-319-65304-4 is OK
- 10.2202/1557-4679.1356 is OK
- 10.1111/rssb.12362 is OK
- 10.1111/biom.13375 is OK
- 10.21105/joss.02526 is OK
- 10.5281/zenodo.1342293 is OK
- 10.5281/zenodo.3558313 is OK
- 10.5281/zenodo.3698329 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1786

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1786, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
Ok I have looked through and I see that the version is up to date, the zenodo archive exists and metadata matches the paper, and the paper reads through ok enough. We can proceed.

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1313952134954708995) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1787
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02447](https://doi.org/10.21105/joss.02447)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats @nhejazi!

Thanks so much to editor @marcosvital and reviewers @klmedeiros and @joethorley for your time, hard work, and expertise. We couldn't do this process without you!

(I will close this issue once the doi resolves)

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02447/status.svg)](https://doi.org/10.21105/joss.02447)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02447">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02447/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02447/status.svg
   :target: https://doi.org/10.21105/joss.02447
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02447/status.svg)](https://doi.org/10.21105/joss.02447)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
